plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,800 @@
+        node_group=dict(type="str", required=True),
+        ipv4_address=dict(type="str", aliases=["peer_address_ipv4"]),
+        ipv6_address=dict(type="str", aliases=["peer_address_ipv6"]),
+        remote_asn=dict(type="int"),
+        admin_state=dict(type="str", choices=["enabled", "disabled"]),
+        import_route_map=dict(type="str"),
+        export_route_map=dict(type="str"),
+        peer_prefix=dict(type="str"),
+        import_route_map_uuid=dict(type="str"),
+        export_route_map_uuid=dict(type="str"),
+        peer_prefix_uuid=dict(type="str"),
+        ebgp_multi_hop_ttl=dict(type="int"),
+        auth_password=dict(type="str", no_log=True),
+        weight=dict(type="int"),
+        site_of_origin=dict(type="str", aliases=["fabric_of_origin"]),
+        allowed_self_as_count=dict(type="int"),
+        local_asn_config=dict(type="str", choices=list(LOCAL_ASN_CONFIG)),
+        local_asn=dict(type="int"),
+        bgp_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                allow_self_as=dict(type="bool"),
+                override_as=dict(type="bool"),
+                disabled_peer_as_check=dict(type="bool"),
+                next_hop_self=dict(type="bool"),
+                send_community=dict(type="bool"),
+                send_extended_community=dict(type="bool"),
+                send_domain_path=dict(type="bool"),
+            ),
+        ),
+        peer_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                bfd=dict(type="bool"),
+                disable_peer_connected_check=dict(type="bool"),
+            ),
+        ),
+        address_families=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                multicast=dict(type="bool"),
+                unicast=dict(type="bool"),
+            ),
+        ),
+        private_as_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                remove_all=dict(type="bool"),
+                replace_with_local_as=dict(type="bool"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["template", "template_id"], True],
+            ["state", "query", ["template", "template_id"], True],
+            ["state", "absent", ["template", "template_id"], True],
+            ["state", "present", ["l3out", "l3out_uuid"], True],
+            ["state", "query", ["l3out", "l3out_uuid"], True],
+            ["state", "absent", ["l3out", "l3out_uuid"], True],
```

Comment:
Is there a reason to do this over required_one_of and mutually_exclusive? I understand the required_if works but from code reading it might be easier to understand.

In ansible docs https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec-dependencies the following is mentioned: n this example, at least one of path and content must be specified. If none are specified, execution will fail. Specifying both is explicitly allowed; to prevent this, combine required_one_of with mutually_exclusive.